docs/quickstart_batch_job.md

```diff
@@ -0,0 +1,201 @@
+
+The Bodywork project for this single-stage workflow is packaged as a [GitHub repository](https://github.com/bodywork-ml/bodywork-batch-job-project), whose root directory is structured as follows,
+
+```text
+root/
+ |-- score-data/
+     |-- score.py
+     |-- requirements.txt
+     |-- config.ini
+     |-- classification_model.joblib
+ |-- bodywork.ini
+```
+
+## Configuring the Bodywork Batch Stage
+
+The `score-data` directory contains the code and configuration required to run the ML job within a pre-built container on a k8s cluster, as a batch workload. The `score.py` module is a standalone and executable Python module that contains the code required to:
+
+* download the new dataset from cloud storage (AWS S3);
+* load the pre-trained model `classification_model.joblib`;
+* score the dataset dataset; and,
+* save the results back to cloud storage (AWS S3).
+
+It can be summarised as,
+
+```python
+from urllib.request import urlopen
+
+# other imports
+# ...
+
+DATA_URL = ('http://bodywork-batch-job-project.s3.eu-west-2.amazonaws.com'
+            '/data/iris_classification_data.csv')
+
+# other constants
+# ...
+
+
+def main() -> None:
+    """Main script to be executed."""
+    data = download_dataset(DATA_URL)
+    model = load(TRAINED_MODEL_FILENAME)
+    scored_data = score_data(data, model)
+    upload_results(scored_data)
+
+
+# other functions definitions used in main()
+# ...
+
+
+if __name__ == '__main__':
+    main()
+```
+
+We recommend that you spend five minutes familiarising yourself with the full contents of [score.py](https://github.com/bodywork-ml/bodywork-batch-job-project/blob/master/score-data/score.py). When Bodywork runs the stage, it will do so in exactly the same way as if you were to run,
+
+```shell
+$ python score.py
+```
+
+And so everything defined in `main()` will be executed.
+
+The `requirements.txt` file lists the 3rd party Python packages that will be Pip-installed on the pre-built Bodywork container, as required to run the `score.py` script. In this example we have,
+
+```text
+boto3==1.16.15
+joblib==0.17.0
+pandas==1.1.4
+scikit-learn==0.23.2
+```
+
+* `boto3` - for interacting with AWS;
+* `joblib` - for persisting models;
+* `pandas` - for manipulating the raw data; and,
+* `scikit-learn` - for training the model.
+
+Finally, the `config.ini` file allows us to configure the key parameters for the stage,
+
+```ini
+[default]
+STAGE_TYPE="batch"
+EXECUTABLE_SCRIPT="score.py"
+CPU_REQUEST=0.5
+MEMORY_REQUEST_MB=100
+
+[batch]
+MAX_COMPLETION_TIME_SECONDS=30
+RETRIES=2
+```
+
+From which it is clear to see that we have specified that this stage is a batch stage (as opposed to a service-deployment), that `score.py` should be the script that is run, together with an estimate of the CPU and memory resources to request from the k8s cluster, how long to wait and how many times to retry, etc.
```

Comment:
If I remember correctly, defaults are used if these are not specified. If so then specify that defaults will be used if they're not present.

Reply:
Just checked the source code - no defaults are used.